jdi-light/src/test/java/mockitotests/HasAttrMockTests.java

```diff
@@ -0,0 +1,87 @@
+
+import static org.mockito.Mockito.mock;
+import static org.testng.Assert.assertTrue;
+
+public class HasAttrMockTests {
+
+    private UIElement uiElement = null;
+    private static final String exceptionMessageFirstPart = "Expected: an array containing";
+    private static final String exceptionMessageSecondPart = "but: mismatches were: [";
+
+    private MapArray<String, String> attributesSet1 = new MapArray<>(new String[]{"type", "value", "class"},
+            new String[]{"checkbox", "option1", "form-check"});
+    private MapArray<String, String> attributesSet2 = new MapArray<>(new String[]{"type", "value", "class"},
+            new String[]{"checkbox", "option1", "form-check"});
+    private MapArray<String, String> attributesSet3 = new MapArray<>(new String[]{"type", "value", "class"},
+            new String[]{"checkbox", "option1", "form-check"});
+    private MapArray<String, String> attributesSet4 = new MapArray<>(new String[]{"type", "value", "class"},
+            new String[]{"checkbox", "option1", "form-check"});
+
+
+    @DataProvider
+    public Object[][] attributesPositive() {
+        return new Object[][]{
+                {attributesSet1, "type"},
+                {attributesSet2, "value"},
+                {attributesSet3, "class"},
+                {attributesSet4, "class"}
+        };
+    }
+
+    @DataProvider
+    public Object[][] attributesNegative() {
+        return new Object[][]{
+                {attributesSet1, "type1"},
+                {attributesSet2, "value1"},
+                {attributesSet3, "class1"},
+                {attributesSet4, "class1"}
+        };
+    }
+
+    @BeforeMethod
+    public void init() {
+        uiElement = mock(UIElement.class);
+    }
+
+    @Test(expectedExceptions = {RuntimeException.class}, dataProvider = "attributesNegative",
+            expectedExceptionsMessageRegExp = ".*Expected: an array containing.*[a-zA-Z'].*but: mismatches were:.*[a-zA-Z'].*")
+    public void hasAttrNegativeOuterExceptionHandlingTest(MapArray<String, String> actualMapAttr, String wrongAttr) {
+        uiElement = Mockito.spy(new UIElement());
+        Mockito.doReturn(actualMapAttr).when(uiElement).attrs();
+        uiElement.is().hasAttr(wrongAttr);
+    }
+
+    @Test(dataProvider = "attributesNegative")
+    public void hasAttrNegativeInnerExceptionHandlingTest(MapArray<String, String> actualMapAttr, String wrongAttr) {
+        uiElement = Mockito.spy(new UIElement());
+        Mockito.doReturn(actualMapAttr).when(uiElement).attrs();
+
+        try {
+            uiElement.is().hasAttr(wrongAttr);
+        } catch (RuntimeException exp) {
+            String expMessage = exp.getMessage();
+            assertTrue(expMessage.contains(exceptionMessageFirstPart + " \"" + wrongAttr + "\""));
+            assertTrue(expMessage.contains(exceptionMessageSecondPart));
+            for (String key : actualMapAttr.keys()) {
+                System.out.println("was  \"" + key + "\"");
```

Comment:
please remove this log

Reply:
@mf-code Fixed , sorry forgot about that :)
I checked all requirements in PR review file, it fits, but assertContains needed to be used (not JDI assert)
And Travis is still falling